scripts/oec_data_source_script.rb

```diff
@@ -0,0 +1,154 @@
+require 'selenium-webdriver'
+
+begin
+
+  puts "#{Time.now} - Beginning OEC data source update script"
+  args = ARGV.to_s
+  campus_data_sources = %w(Campus_Courses Campus_Supervisors Campus_Instructors Campus_Students Campus_Course_Supervisors Campus_Course_Instructors Campus_Course_Students)
+  extension_data_sources = Dir[File.join(ENV['HOME'], 'selenium-files/*.csv')]
+
+  @base_url = args.include?('qa') ? 'https://course-evaluations-qa.berkeley.edu' : 'https://course-evaluations.berkeley.edu'
+
+  # Set timeouts and launch browser
+  @brief_wait = Selenium::WebDriver::Wait.new(timeout: 30)
+  @medium_wait = Selenium::WebDriver::Wait.new(timeout: 120)
+  @long_wait = Selenium::WebDriver::Wait.new(timeout: 900)
+  @driver = Selenium::WebDriver.for :firefox
+  @driver.manage.window.maximize
+
+  def click_element_id(driver, id)
+    @brief_wait.until { driver.find_element(id: id) }
+    driver.find_element(id: id).click
+  end
+
+  def log_in
+    puts "#{Time.now} - Logging in to Blue"
+    @driver.get "#{@base_url}"
+    click_element_id(@driver, 'username')
+    @long_wait.until { @driver.find_element(id: 'BlueAppControl_admin-link-btn') }
+  end
+
+  def find_and_edit_source(data_source)
+    # Load homepage, click Admin button, and click Data Sources
+    puts "#{Time.now} - Processing '#{data_source}'"
+    @driver.get "#{@base_url}"
+    click_element_id(@driver, 'BlueAppControl_admin-link-btn')
+    click_element_id(@driver, 'AdminUC_menu_item_data_sources')
+    # Search for the data source type
+    puts "#{Time.now} - Searching for data source"
+    @brief_wait.until { @driver.find_element(id: 'AdminUC_DataSources_AdminDataSource_Tabs_MultiDataSource_listing') }
+    sleep 2
+    @driver.find_element(id: 'AdminUC_DataSources_AdminDataSource_Tabs_tbSearchValue').send_keys data_source
+    @driver.find_element(id: 'AdminUC_DataSources_AdminDataSource_Tabs_btnSearch').click
+    @brief_wait.until { @driver.find_element(xpath: "//table[@id='AdminUC_DataSources_AdminDataSource_Tabs_MultiDataSource_listing']//tr[3][contains(.,'#{data_source}')]") }
+    @driver.find_element(link_text: 'Edit').click
+    # Click Data tab and click Edit on Data Blocks tab
+    puts "#{Time.now} - Found data source, editing it"
+    click_element_id(@driver, 'AdminUC_Data_Data')
+    @brief_wait.until do
+      @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_lblDataSource')
+      @brief_wait.until { @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_lblDataSource').text.include? data_source }
```

Comment:
Yes, @johncrossman.  You can't invoke #text if the element is still nil.  And the nested wait is to ensure the text has time to load in the element.